vermouth/molecule.py

```diff
@@ -1105,13 +1106,58 @@ def __init__(self, incoming_graph_data=None, **attr):
 
     def __eq__(self, other):
         return (super().__eq__(other)
-                and set(self.non_edges) == set(other.non_edges)
+                and self._same_non_edges(other)
                 and self.removed_interactions == other.removed_interactions
                 and self.molecule_meta == other.molecule_meta
                 and self.patterns == other.patterns
                 and set(self.features) == set(other.features)
                 )
 
+    def _same_non_edges(self, other):
+        """
+        Returns `True` if all the non-edges of an `other` link are equal to
+        those of this link. Returns `False` otherwise.
+        """
+        # A non-edge is a list of two elements: the key to a node in the graph
+        # that is used as achor, and a attribute dict that must match none of
+        # the atoms connected to the anchor.
+        # For the link to match, none of the non-edges must match. Therefore,
+        # their order do not matter. Though, because the attribute dicts are
+        # not hashable, non-edges cannot be put in a set.
```

Comment:
You can make a frozenset of the dict items.

Reply:
You cannot put a list or a dict in a `frozenset`.